timelock-impl/src/main/java/com/palantir/atlasdb/timelock/watch/LockWatch.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.watch;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.common.collect.Maps;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.lock.LockDescriptor;
+
+class LockWatch {
+    private Map<LockDescriptor, LockIndexState> indexStates;
+    private Map<LockDescriptor, AtomicLong> counters;
```

Comment:
it feels as if you actually want to have one global `Map<LockDescriptor, LockIndexState>` and `Map<LockDescriptor, AtomicLong>`, and then query based on watch ids, where you reconstruct based on the global map. I suppose I'm not super clear how this will be consumed just yet.

`computeIfPresent` also runs atomically, so there doesn't seem to be a need to keep separate maps here, since you only ever update one when updating the other.